### PR TITLE
UI/15498445 buttons

### PR DESCRIPTION
**Pull request by KatarzynaStefanowiczPagepro**
## Description

- [] TODO write description

**New stuff** ✨

* New stuff goes here

**Changes** 🏗

* Changes go here

**Deletions** ⚰️

* Deletions go here

## TODO

- [ ] Add TODOs

(Resolves | Contributes to) #31415

### CODE REVIEW

**Comment by KatarzynaStefanowiczPagepro** [comment]
https://tw.auroracreation.com/app/tasks/15498445

Prepared buttons
image

image

You can find all of the buttons in storybook

File: src/components/common/Extensions/CalamityBanner/CalamityBanner.tsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ const CalamityBanner: FC<PropsWithChildren<CalamityBannerProps>> = ({
               <Button
                 type="button"
                 className="md:mr-[1.6875rem] ml-4"
-                mode="primaryOutline"
+                mode="primaryOutlineFulled"
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/CalamityBanner/CalamityBanner.tsx:48]: it doesn't sounds good, why this change?

File: src/components/common/Extensions/UserNavigation/UserNavigation.tsx [hunk 2]
```diff
@@ -103,46 +102,32 @@ const UserNavigation: FC = () => {
...
           isFullRounded
           setTriggerRef={setWalletTriggerRef}
           onClick={connectWallet}
-          className={clsx('md:border-gray-200 md:hover:border-blue-400', {
-            'px-4 py-2.5 border-base-white text-gray-400':
-              isWalletVisible && isMobile,
-            'p-0': !isWalletVisible && isMobile,
+          iconName={isWalletVisible && isMobile ? 'close' : 'cardholder'}
+          className={clsx({
+            '!border-base-white': isWalletVisible && isMobile,
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/UserNavigation/UserNavigation.tsx:111]: '!border-base-white' is it correct?

File: src/components/v5/common/EmptyContent/EmptyContent.tsx [hunk 3]
```diff
@@ -39,12 +39,15 @@ const EmptyContent: FC<EmptyContentProps> = ({
...
           // @TODO: Change to Button component
-          <button type="button" className="mt-4" onClick={onClick}>
-            {withButtonIcon && (
-              <Icon name="share-network" appearance={{ size: 'normal' }} />
-            )}
+          <Button
+            mode="primaryOutline"
+            className="mt-4"
+            size="small"
+            iconName="share-network"
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/EmptyContent/EmptyContent.tsx:46]: what if there will be a case with a button without an icon?

File: src/components/v5/shared/Button/Button.tsx [hunk 4]
```diff
@@ -13,6 +13,7 @@ const displayName = 'v5.Button';
```
Comment (krawczykkatarzyna) [inline src/components/v5/shared/Button/Button.tsx:13]: because we have a lot of cases to handle is hard to read that file, maybe better solution will be create more then one button? They can depends on their destiny

Comment (krawczykkatarzyna) [reply src/components/v5/shared/Button/Button.tsx:13]: or create "base button" and try to build others button base on that

File: src/components/common/Extensions/UserNavigation/UserNavigation.tsx (hunk 2 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/common/Extensions/UserNavigation/UserNavigation.tsx:111]: Yes, we can't override this style in tailwind in different way in this case :/

File: src/components/v5/common/EmptyContent/EmptyContent.tsx (hunk 3 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/v5/common/EmptyContent/EmptyContent.tsx:46]: You can use:
- only label
- only icon
- both at the same time

If you don't use anything, the button will be empty
image

File: src/components/v5/shared/Button/Button.tsx (hunk 4 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/v5/shared/Button/Button.tsx:13]: I think the text button could be separate.
I'll do it.

Comment (KatarzynaStefanowiczPagepro) [reply src/components/v5/shared/Button/Button.tsx:12]: Done.

File: src/components/common/Extensions/CalamityBanner/CalamityBanner.tsx (hunk 1 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/common/Extensions/CalamityBanner/CalamityBanner.tsx:48]: Primary solid:
image

Primary outline:
image

Primary outline fulled (notice that hover state is fully bg blue):
image

Each of these buttons can have an icon and size default or small.
Do you have any idea how we can name the current 'primaryOutlineFulled' button better?

File: src/components/v5/shared/Button/TextButton.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
+/* eslint-disable react/button-has-type */
+import React, { FC, PropsWithChildren } from 'react';
+import { useIntl } from 'react-intl';
+import clsx from 'clsx';
+
+import { TextButtonProps } from './types';
+import SpinnerLoader from '~shared/Preloaders/SpinnerLoader';
+import styles from './TextButton.module.css';
+
+const displayName = 'v5.Button.TextButton';
```
Comment (krawczykkatarzyna) [inline src/components/v5/shared/Button/TextButton.tsx:10]: v5.shared.Button.TextButton

File: src/components/v5/shared/Button/PendingButton.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+import React, { FC, PropsWithChildren } from 'react';
+import { useIntl } from 'react-intl';
+import clsx from 'clsx';
+
+import { PendingButtonProps } from './types';
+import SpinnerLoader from '~shared/Preloaders/SpinnerLoader';
+import styles from './PendingButton.module.css';
+import Icon from '~shared/Icon';
+
+const displayName = 'v5.Button.PendingButton';
```
Comment (krawczykkatarzyna) [inline src/components/v5/shared/Button/PendingButton.tsx:11]: v5.shared.Button.PendingButton

File: src/components/v5/shared/Button/Button.tsx [hunk 7]
```diff
@@ -13,6 +12,7 @@ const displayName = 'v5.Button';
```
Comment (krawczykkatarzyna) [inline src/components/v5/shared/Button/Button.tsx:12]: v5.shared.Button

File: src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx [hunk 8]
```diff
@@ -68,67 +68,77 @@ const TransactionsItem: FC<TransactionsItemProps> = ({
...
+                    key={key}
+                    className={clsx(styles.listItem, {
+                      'font-semibold': isCurrentAction,
+                      'before:bg-success-400':
+                        isCurrentAction &&
+                        (statusContent === TRANSACTION_STATUSES.READY ||
+                          statusContent === TRANSACTION_STATUSES.SUCCEEDED),
+                      'before:bg-negative-400':
+                        isCurrentAction &&
+                        status === TRANSACTION_STATUSES.FAILED,
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx:92]: status === TRANSACTION_STATUSES.FAILED  are you sure is that correct?

File: src/components/v5/shared/Button/TextButton.tsx (hunk 5 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/v5/shared/Button/TextButton.tsx:10]: As I can see, we have never used the prefix `shared` for components inside this directory.

File: src/components/v5/shared/Button/PendingButton.tsx (hunk 6 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/v5/shared/Button/PendingButton.tsx:11]: As I can see, we have never used the prefix `shared` for components inside this directory.

File: src/components/v5/shared/Button/Button.tsx (hunk 7 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/v5/shared/Button/Button.tsx:12]: As I can see, we have never used the prefix `shared `for components inside this directory.

File: src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx (hunk 8 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx:92]: I don't know. I've only destructured this object to be better readable, I didn't check conditionals.
image
We can check it and fix it in a separate PR if it needed

Comment (krawczykkatarzyna) [reply src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx:92]: `statusContent` not `status`

Comment (KatarzynaStefanowiczPagepro) [reply src/components/common/Extensions/UserHub/partials/TransactionsTab/partials/TransactionsItem/TransactionsItem.tsx]: Fixed

**Review by krawczykkatarzyna — APPROVED** [review]
